src/main/java/com/coveo/spillway/Spillway.java

```diff
@@ -121,14 +121,43 @@ public boolean tryCall(T context) {
    *
    * @param context Either the name of the limit OR the object on which the propertyExtractor ({@link LimitBuilder#of(String, java.util.function.Function)})
    *                will be applied if it was specified
-   * @param cost The cost of the query
-   * @return False if one the enforced limit is exceeded, true otherwise
+   * @param cost The cost of the query, must be positive
+   * @return False if one the enforced limits is exceeded, true otherwise
    */
   public boolean tryCall(T context, int cost) {
-    return getExceededLimits(context, cost).isEmpty();
+    return getExceededLimits(context, cost, true).isEmpty();
   }
 
-  private List<LimitDefinition> getExceededLimits(T context, int cost) {
+  /**
+   * Behavess like {@link #checkLimit(Object, int)} with {@code cost} of one.
+   *
+   * @see #checkLimit(Object, int)
+   *
+   * @param context Either the name of the limit OR the object on which the propertyExtractor ({@link LimitBuilder#of(String, java.util.function.Function)})
+   *                will be applied if it was specified
+   * @return False if one the enforced limits would be exceeded, true otherwise
+   */
+  public boolean checkLimit(T context) {
+    return checkLimit(context, 1);
+  }
+
+  /**
+   * Verify if the given limit will be exceeded for the given cost
+   *
+   * @param context Either the name of the limit OR the object on which the propertyExtractor ({@link LimitBuilder#of(String, java.util.function.Function)})
+   *                will be applied if it was specified
+   * @param cost The cost of the query, must be positive
+   * @return False if one the enforced limits would be exceeded, true otherwise
+   */
+  public boolean checkLimit(T context, int cost) {
+    return getExceededLimits(context, cost, false).isEmpty();
+  }
+
+  private List<LimitDefinition> getExceededLimits(T context, int cost, boolean performAdd) {
```

Comment:
adding a boolean parameter is less than ideal, but I also recognize wanting to introduce a minimum of change here - this is a pretty critical lib used in a bunch of places.

that said, any way to split this up into separate functions instead? not a big deal though.

Reply:
Well, if we were moving the trigger handling, that would make it easy to do, if not, it gets a bit messy